tests/DataTransferObjectTest.php

```diff
@@ -36,6 +36,24 @@ public function only_the_type_hinted_type_may_be_passed()
         };
     }
 
+    /** @test */
+    public function invalid_types_set_exception_properties()
+    {
+        try {
+            new class(['foo' => 'value']) extends DataTransferObject {
+                /** @var boolean */
+                public $foo;
+            };
+        } catch (DataTransferObjectError $error) {
```

Comment:
Just a heads-up: if the library code changes and throws no exception, it might go unnoticed here (might still show up as a test having no assertions though).

I suggest a different approach (for this and the other tests):
```php
$error = null;
try {
  …
} catch (…) {
  // Deliberately empty
}
$this->asssertInstanceOf(DataTransferObjectError::class, $error);
$this->assertEquals(…)
```

Reply:
I used an existing test as the template for my new tests. You're definitely right - this creates the possibility of hiding assertions if the throws change. Great catch, thank you!